src/ivectorbin/ivector-compute-lda.cc

```diff
@@ -84,11 +84,23 @@ template<class Real>
 void ComputeNormalizingTransform(const SpMatrix<Real> &covar,
                                  MatrixBase<Real> *proj) {
   int32 dim = covar.NumRows();
-  TpMatrix<Real> C(dim);  // Cholesky of covar, covar = C C^T
-  C.Cholesky(covar);
-  C.Invert();  // The matrix that makes covar unit is C^{-1}, because
-               // C^{-1} covar C^{-T} = C^{-1} C C^T C^{-T} = I.
-  proj->CopyFromTp(C, kNoTrans);  // set "proj" to C^{-1}.
+  Matrix<Real> U(dim, dim);
+  Vector<Real> s(dim);
+  covar.Eig(&s, &U);
+  // Sort from greatest to smallest eigenvalue.
+  SortSvd(&s, &U);
+  // Floor eigenvalues to a small positive value.
+  Real floor = s(0) * std::numeric_limits<Real>::min();
```

Comment:
@david-ryan-snyder, this  floor is too small.. the problem is you take the square root and invert it before constructing the LDA matrix, and certain directions will be too large-- for roundoff reasons, if nothing else.  I suggest to make it configurable, with a default of 0.0001.  This will likely never make a difference for Kaldi recipes because our i-vectors and x-vectors have a reasonable distribution.   It may affect other scenarios, though.
(note: if this was set to 1.0, we'd have PCA, not LDA.)